searching/binarysearch.c

```diff
@@ -0,0 +1,31 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+
+int binarySearch(int vet[], int k, int n){ 
+    int l = 0, r = n-1, m;
+    while(l<=r){
+        m = (l+r)/2;
+        if(k == vet[m]){
+            return m;
+        }else if(k < vet[m]){
+            r = m-1;
+        }else{
+            l = m+1;
+        }
+    }
+    return -1;
+}
+
+int main(){
+    
+    int vet[] = {1,2,3,4,5,6,7,8,9,10}, res; //Here I create a array called vet, with ten positions, and a variable called res
+    
+    res = binarySearch(vet, 5, 9); //Now I call my binarySearch function, I send to my function my array called vet, the number that I wanna find
```

Comment:
You should be calling the function with 10 as the last argument, not 9 (`binarySearch(vet, 5, 9)`)

Reply:
For sure, I didn't saw this, but now I'm passing 10 as the array size, thank u.